apps/devhub/tests/test_views_versions.py

```diff
@@ -186,6 +187,20 @@ def test_user_can_disable_addon(self, hide_mock):
         msg = entry.to_string()
         assert self.addon.name.__unicode__() in msg, ("Unexpected: %r" % msg)
 
+    def test_user_can_unlist_addon(self):
+        self.addon.update(status=amo.STATUS_PUBLIC, disabled_by_user=False,
+                          is_listed=True)
+        res = self.client.post(self.unlist_url)
+        eq_(res.status_code, 302)
```

Comment:
That the difficult part: striking a balance between convention/coherence and the "better way". I try to use the pytest way whenever I can, meaning whenever I'm writing a new test (or rewriting most of an old one, in which case i rewrite it entirely).

It's open to discussion though.